### PR TITLE
Storage network on controllers, disable dhcp on unused interfaces

### PR DESCRIPTION
**Pull request by dmi-try**

### CODE REVIEW

**Comment by mihgen** [comment]
+1

File: lib/astute/deployment_engine.rb [hunk 1]
```diff
@@ -74,15 +75,18 @@ def attrs_ha(nodes, attrs)
                    n['network_data'].select {|nd| nd['name'] == 'management'}[0]['ip'].split(/\//)[0]})
         ctrl_public_addrs.merge!({hostname =>
                    n['network_data'].select {|nd| nd['name'] == 'public'}[0]['ip'].split(/\//)[0]})
+        ctrl_storage_addrs.merge!({hostname =>
+                   n['network_data'].select {|nd| nd['name'] == 'storage'}[0]['ip'].split(/\//)[0]})
```
Comment (AlexanderPavlenko) [inline lib/astute/deployment_engine.rb:79]: `split(/\//) == split('/')`

File: lib/astute/deployment_engine/nailyfact.rb [hunk 2]
```diff
@@ -10,7 +10,8 @@ def create_facts(node, attrs)
     # if node['network_data'] is undefined, we use empty list because we later try to iterate over it
     #   otherwise we will get KeyError
     node_network_data = node['network_data'].nil? ? [] : node['network_data']
-    network_data_puppet = calculate_networks(node_network_data)
+    interfaces = node['meta']['interfaces']
```
Comment (AlexanderPavlenko) [inline lib/astute/deployment_engine/nailyfact.rb:13]: what's inside? something like `[{'name' => …}, … ]`
tests are broken

Comment (dmi-try) [inline lib/astute/deployment_engine/nailyfact.rb:13]:
```
        "interfaces": [
            {
                "mac": "52:54:00:04:A9:02", 
                "name": "eth1", 
                "current_speed": null
            }, 
            {
                "ip": "10.0.76.130", 
                "mac": "52:54:00:C6:85:C7", 
                "netmask": "255.255.255.0", 
                "current_speed": null, 
                "name": "eth0"
            }
```

**Pull request merged**